internal/common/lxd_interface.go

```diff
@@ -0,0 +1,111 @@
+package common
+
+import (
+	"context"
+	"strings"
+
+	"github.com/canonical/lxd/shared/api"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Interface represents LXD instance network interface.
+type InterfaceModel struct {
+	// Real name of the interface within the instance. If config interface is
+	// defined as "eth0", the real interface within a container will have the
+	// same name. However, VMs will most likely have enp0sX.
```

Comment:
For VMs, it's usually `enpXs0` where X starts at 5.

Reply:
Thanks, will fix :)